### PR TITLE
Added try-except clause to catch numpy error.

### LINKED ISSUE #27953: DataFrame constructor fails with extension dtype and columns
I *think* this should be supported.

```python
In [8]: pd.DataFrame(columns=['a'], dtype=pd.CategoricalDtype())
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-8-a44ecacab92a> in <module>
----> 1 pd.DataFrame(columns=['a'], dtype=pd.CategoricalDtype())

~/sandbox/pandas/pandas/core/frame.py in __init__(self, data, index, columns, dtype, copy)
    405             )
    406         elif isinstance(data, dict):
--> 407             mgr = init_dict(data, index, columns, dtype=dtype)
    408         elif isinstance(data, ma.MaskedArray):
    409             import numpy.ma.mrecords as mrecords

~/sandbox/pandas/pandas/core/internals/construction.py in init_dict(data, index, columns, dtype)
    224         # no obvious "empty" int column
    225         if missing.any() and not is_integer_dtype(dtype):
--> 226             if dtype is None or np.issubdtype(dtype, np.flexible):
    227                 # GH#1783
    228                 nan_dtype = object

~/Envs/pandas-dev/lib/python3.7/site-packages/numpy/core/numerictypes.py in issubdtype(arg1, arg2)
    391     """
    392     if not issubclass_(arg1, generic):
--> 393         arg1 = dtype(arg1).type
    394     if not issubclass_(arg2, generic):
    395         arg2_orig = arg2

TypeError: data type not understood
```

**Expected Output**

```python
In [10]: pd.DataFrame(columns=['a']).astype('category')
Out[10]:
Empty DataFrame
Columns: [a]
Index: []
```

### PR DESCRIPTION
**Pull request by dustanlevenstein**
- [x] closes #27953
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jbrockmendel** [comment]
needs a test

**Review by jreback — CHANGES_REQUESTED** [review]
always add tests first
we are also trying to remove try/excepts except in very limited circumstances as they hide errors

**Comment by dustanlevenstein** [comment]
Modified as requested.

File: pandas/core/internals/construction.py [hunk 1]
```diff
@@ -223,7 +224,9 @@ def init_dict(data, index, columns, dtype=None):
 
         # no obvious "empty" int column
         if missing.any() and not is_integer_dtype(dtype):
-            if dtype is None or np.issubdtype(dtype, np.flexible):
+            if isinstance(dtype, CategoricalDtype):
```
Comment (jbrockmendel) [inline pandas/core/internals/construction.py]: this check would use `is_categorical_dtype(dtype)`

**Comment by jbrockmendel** [comment]
by "needs a test" I meant that this PR fixes some bug and we need a unit test (goes somewhere in pandas/tests/) to test that we get the correct behavior

**Comment by dustanlevenstein** [comment]
I added a test in a place that seemed appropriate. It passed on my platform.

**Comment by dustanlevenstein** [comment]
The last version didn't pass the checks. I'm confused because the error doesn't seem to be related to anything that I edited...

Screenshot from 2019-08-18 15-06-44
Screenshot from 2019-08-18 15-05-19

File: pandas/tests/frame/test_dtypes.py [hunk 2]
```diff
@@ -69,6 +69,10 @@ def test_empty_frame_dtypes_ftypes(self):
         norows_df = pd.DataFrame(columns=list("abc"))
         assert_series_equal(norows_df.dtypes, pd.Series(np.object, index=list("abc")))
 
+        cat = pd.CategoricalDtype()
```
Comment (mroeschke) [inline pandas/tests/frame/test_dtypes.py]: Could you add tests for other extension dtypes (e.g. datetime with timezones and/or period)

File: pandas/core/internals/construction.py [hunk 3]
```diff
@@ -223,7 +223,9 @@ def init_dict(data, index, columns, dtype=None):
 
         # no obvious "empty" int column
         if missing.any() and not is_integer_dtype(dtype):
-            if dtype is None or np.issubdtype(dtype, np.flexible):
+            if is_categorical_dtype(dtype):
```
Comment (mroeschke) [inline pandas/core/internals/construction.py]: Based on the original issue this needs to be `is_extension_dtype`.

**Review by mroeschke — CHANGES_REQUESTED** [review]
Also needs a whatsnew entry in 1.0.0

**Comment by jreback** [comment]
can you merge master and add a test

**Comment by jreback** [comment]
closing this as stale, though its a nice change, needs a rebase and comments addressed.

**Pull request closed without merging**